server/neptune/gateway/event/preworkflow/hook_executor.go

```diff
@@ -0,0 +1,64 @@
+package preworkflow
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"os"
+	"os/exec"
+)
+
+// interface for actual execution makes local testing easier
+
+type HookExecutor interface {
+	Execute(hook *valid.PreWorkflowHook, repo models.Repo, path string) error
+}
+
+type PreWorkflowHookExecutor struct {
+}
+
+func (e *PreWorkflowHookExecutor) Execute(hook *valid.PreWorkflowHook, repo models.Repo, path string) error {
```

Comment:
do we plan to use the legacy repo model here?

Reply:
yes for now, everything on the gateway side still is using the legacy repo